src/dp_flow.c

```diff
@@ -294,6 +301,7 @@ int dp_get_flow(const struct flow_key *key, struct flow_value **p_flow_val)
 bool dp_are_flows_identical(const struct flow_key *key1, const struct flow_key *key2)
 {
 	return key1->proto == key2->proto
+		// this should compare ipv6 too
 		&& key1->l3_src.ip4 == key2->l3_src.ip4
 		&& key1->l3_dst.ip4 == key2->l3_dst.ip4
 		&& key1->port_dst == key2->port_dst
```

Comment:
Here I actually noticed that we are manually comparing every field of the key anyway. So I propose to move this into a `static __rte_always_inline` and compare it there in a fast way (using `uint64_t` comparisons). Because that is the fastest way to compare fixed-size blocks (for variable-size `rte_memcmp` is best).

Reply:
Good idea.